### ISSUE TITLE
Terrafrom appears to ignore aws config

### ISSUE THREAD

**Issue by kneufeld-pbp** [hashicorp/terraform-provider-aws#14396]
We are experiencing a Terraform authentication issue; our assumption is that Terraform knows how to utilize the AWS advance settings in the `config` file just like `aws cli`. Specifically, when executing from an EC2 instance with an IAM_ROLE that can assume other roles.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
```
Terraform v0.12.19
+ provider.aws v2.70.0

Your version of Terraform is out of date! The latest version
is 0.12.29. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)
- terraform
- aws-provider?

##### main.tf:
```hcl
provider "aws" {
    region = "us-west-2"
}

data "aws_caller_identity" "current" {}

output "account_id" {
  value = "${data.aws_caller_identity.current.account_id}"
}

output "caller_arn" {
  value = "${data.aws_caller_identity.current.arn}"
}

output "caller_user" {
  value = "${data.aws_caller_identity.current.user_id}"
}
```

##### AWS Config:
Location: `~/.aws/config`
```
[profile sandbox]
role_arn=arn:aws:iam:...
credential_source=Ec2InstanceMetadata
```  
We have also used the `AWS_CONFIG_FILE` variable to override the location when needed.

#####  Environment Variables SET
AWS_CONFIG_FILE = ~/.aws/config
AWS_PROFILE = sandbox
AWS_SDK_LOAD_CONFIG = 1

### Expected Behavior

Should get the same caller identity as `aws sts get-caller-identity` the sandbox profile 

### Actual Behavior
- `aws sts get-caller-identity` outputs the correct sandbox profile
```json
{
    "UserId": "ARO...:botocore-session-1496068120",
    "Account": "XXXXXXXXXX2",
    "Arn": "arn:aws:sts::XXXXXXXXXX2:assumed-role/SandboxRole/botocore-session-1496068120"
}
```

- terraform outputs the assigned EC2 IAM role
```
account_id = XXXXXXXXXX1
caller_arn = arn:aws:sts::XXXXXXXXXX1:assumed-role/EC2IamRole/i-0ddd...
caller_user = ARO...:i-0ddd...
```

### References

* #0000

**Comment by strowk** [comment]
Same problem for me.
To avoid it with `version = "~> 2.70"`
we had to set `skip_metadata_api_check = true` as explained in https://github.com/terraform-providers/terraform-provider-aws/issues/5018

Unfortunately 3.0 breaks that workaround and does not fix the problem as this comment claims: https://github.com/terraform-providers/terraform-provider-aws/issues/5018#issuecomment-657582432

**Comment by dtw45** [comment]
> Unfortunately 3.0 breaks that workaround and does not fix the problem as this comment claims: #5018 (comment)

Using v3.10.0 I actually noticed that it does adhere to the `profile` and `shared_credentials_file` declared when using local state. 

Starting a new project you can init with local state and apply, which uses the correct credentials without having to set any environment variables. If you then configure remote state and run init, terraform will ignore the config and use your default credentials.

**Comment by justinretzolk** [comment]
Hi @kneufeld-pbp 👋 It looks like there's been some changes around this behavior in more recent versions of the AWS provider. Can you confirm whether you're still experiencing this issue?

**Comment by nevelis** [comment]
@justinretzolk Still a problem in 3.74.0

**Comment by gdavison** [comment]
This was resolved in v4.0 (or earlier) of the AWS provider. If you run into this problem again, please open a new issue.
